### DISCUSSION TITLE
Router doesn't infer types properly

### DISCUSSION THREAD

**Discussion by TenshiMunasinghe** [trpc/trpc#2075] [Q&A] [answered]
Here's my router code.

```ts
import { PrismaClient } from '@prisma/client'
import * as trpc from '@trpc/server'
import * as trpcNext from '@trpc/server/adapters/next'
import z from 'zod'

const prisma = new PrismaClient()

const appRouter = trpc.router().query('login', {
  input: z.object({
    email: z.string(),
    password: z.string(),
  }),
  async resolve({ input }) {
    return prisma.user.findFirst({
      where: {
        email: input.email,
        password: input.password,
      },
    })
  },
})
```

and here's it's type:
```ts
const appRouter: Router<unknown, unknown, {}, Record<"login", Procedure<unknown, unknown, {}, undefined, undefined, User, unknown, User>>, {}, {}, trpc.DefaultErrorShape>
```

As you can see, it doesn't infer the type of request input which is
```ts
{
  email: string
  password: string
}
```
and instead, it's just `undefined`.

What am I doing wrong here?

**Comment by blackdynamo** [comment]
Are you using Typescript? Have you ensured that it's in strict mode?

No strict
![image](https://user-images.githubusercontent.com/245688/176980854-a3173bb0-0d36-4793-a903-442b398ccf3d.png)

Strict mode
![image](https://user-images.githubusercontent.com/245688/176980868-2baa8185-718a-4027-8f89-e2c30ee4aa01.png)

**Comment by sachinraja** [comment]
If you are using strict mode and still seeing the issue, we need a reproduction.

**Comment by TenshiMunasinghe** [comment, marked as answer]
I had two tsconfig.json files. One in the root folder and other inside the trpc folder where I had a file consisting the router code. The strict mode was enabled only in the one inside the trpc folder. Removing tsconfig.json inside trpc folder and enabling strict mode in root tsconfig.json solved the issue.

**Comment by NeurAlch** [comment]
This might be related: https://github.com/trpc/trpc/discussions/2077
